correios_lib/base.py

```diff
@@ -89,11 +120,23 @@ def call(self, method, request):
         '''
         service_method = getattr(self.client.service, method, None)
 
-        try:
-            return service_method(request)
-        except ValueError:
-            raise WebserviceError('The method you are trying to call does not '
-                                  'exists.')
+        req_type = 'Request' + method[0].upper() + method[1:]
+        req_instance = request.__class__.__name__
+
+        if req_type != req_instance:
+            raise WebserviceError(
+                'Request must be an instance of {0}, given {1}'.format(
+                    req_type,
+                    req_instance
+                 )
+            )
+
+        import sys
+        if sys.version_info <= (3, 0):
+            reload(sys)
+            sys.setdefaultencoding('utf-8')
+
+        return service_method(**request)
```

Comment:
you are getting the service_method in a super safe way but then you don't check if it's **None** or not; this will rise something like `TypeError: 'NoneType' object is not callable`